src/data/github/repo.rs

```diff
@@ -0,0 +1,36 @@
+use crate::errors::*;
+
+use miette::{miette, IntoDiagnostic};
+use url::Url;
+
+#[derive(Debug)]
+pub struct GithubRepo {
+    pub owner: String,
+    pub name: String,
+}
+
+impl GithubRepo {
+    pub fn from(repo_url: &str) -> Result<Self> {
+        let repo_parsed = match Url::parse(repo_url).into_diagnostic() {
+            Ok(parsed) => Ok(parsed),
+            Err(e) => Err(OrandaError::RepoParseError {
+                repo: repo_url.to_string(),
+                details: e,
+            }),
+        };
+        let binding = repo_parsed?;
```

Comment:
```suggestion
        let binding = Url::parse(repo_url).into_diagnostic().map_err(|e| OrandaError::RepoParseError {
            repo: repo_url.to_string(),
            details: e,
        })?;
```

nice place for map_err